api/test/e2enew/route/route_with_management_fields_test.go

```diff
@@ -0,0 +1,461 @@
+			Method:       http.MethodGet,
+			Path:         "/apisix/admin/notexist/routes",
+			Query:        "name=jack",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusBadRequest,
+			ExpectBody:   "Route name is reduplicate",
+			Sleep:        base.SleepTime,
+		}),
+		table.Entry("check route exists by name (exclude it self)", base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Method:       http.MethodGet,
+			Path:         "/apisix/admin/notexist/routes",
+			Query:        "name=jack&exclude=r1",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		}),
+		table.Entry("access the route's uri (r1)", base.HttpTestCase{
+			Object:       base.APISIXExpect(),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "hello world",
+		}),
+		table.Entry("verify the route's content (r1)", base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Path:         "/apisix/admin/routes/r1",
+			Method:       http.MethodGet,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "\"name\":\"jack\",\"desc\":\"config route with name and desc\"",
+		}),
+	)
+
+	ginkgo.It("get the route information", func() {
+		time.Sleep(time.Duration(100) * time.Millisecond)
+		basepath := base.ManagerAPIHost + "/apisix/admin/routes"
+		request, _ := http.NewRequest("GET", basepath+"/r1", nil)
+		request.Header.Add("Authorization", base.GetToken())
+		resp, err := http.DefaultClient.Do(request)
+		if err != nil {
+			return
+		}
+		defer resp.Body.Close()
+		respBody, _ := ioutil.ReadAll(resp.Body)
+		createtime = gjson.Get(string(respBody), "data.create_time")
+		updatetime = gjson.Get(string(respBody), "data.update_time")
+		gomega.Expect(createtime.Int()).To(gomega.SatisfyAll(
+			gomega.BeNumerically(">=", time.Now().Unix()-1),
+			gomega.BeNumerically("<=", time.Now().Unix()+1),
+		))
+
+		gomega.Expect(updatetime.Int()).To(gomega.SatisfyAll(
+			gomega.BeNumerically(">=", time.Now().Unix()-1),
+			gomega.BeNumerically("<=", time.Now().Unix()+1),
+		))
+	})
+
+	table.DescribeTable("test for route with name description",
+		func(tc base.HttpTestCase) {
+			base.RunTestCase(tc)
+		},
+		table.Entry("update the route (r1)", base.HttpTestCase{
+			Object: base.ManagerApiExpect(),
+			Path:   "/apisix/admin/routes/r1",
+			Method: http.MethodPut,
+			Body: `{
+					"name": "route1",
+					"uri": "/hello",
+					"name": "new jack",
+					"desc": "new desc",
+					"upstream": {
+						"type": "roundrobin",
+						"nodes": {
+							"` + base.UpstreamIp + `:1980": 1
+						}
+					}
+				}`,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+			Sleep:        time.Duration(2) * time.Second,
```

Comment:
need to check body too

Reply:
okay. Just for this one or all PUT requests

Reply:
We can test the fixed part, example:

```go
ExpectBody: []string{`"code":0`, `"name":"new jack"`},
```

Reply:
Cool. Adding changes.

Reply:
Now, all tests with PUT request have been updated.